lua/mapvote/server/modules/net.lua

```diff
@@ -1,46 +1,72 @@
+MapVote.Net = MapVote.Net or {}
+
+util.AddNetworkString( "MapVote_RequestConfig" )
+util.AddNetworkString( "MapVote_Config" )
+
+util.AddNetworkString( "MapVote_RequestMapList" )
+util.AddNetworkString( "MapVote_MapList" )
+
 util.AddNetworkString( "MapVote_VoteStarted" )
 util.AddNetworkString( "MapVote_VoteCancelled" )
-util.AddNetworkString( "MapVote_ChangeVote" )
 util.AddNetworkString( "MapVote_VoteFinished" )
+
+util.AddNetworkString( "MapVote_ChangeVote" )
 util.AddNetworkString( "MapVote_PlayerChangedVote" )
-util.AddNetworkString( "MapVote_MapList" )
+
 util.AddNetworkString( "RTV_Delay" )
 
 -- from client
-util.AddNetworkString( "MapVote_RequestMapList" )
-util.AddNetworkString( "MapVote_RequestConfig" )
-
-net.Receive( "MapVote_RequestMapList", function( _, ply )
-    if not ply:IsSuperAdmin() then return end
-
-    local mapList = MapVote.getMapList()
-
-    net.Start( "MapVote_MapList" )
-    net.WriteUInt( #mapList, 32 )
-    for _, map in pairs( mapList ) do
-        net.WriteString( map )
-    end
-    net.Send( ply )
-end )
-
-net.Receive( "MapVote_RequestConfig", function( _, ply )
-    if not ply:IsSuperAdmin() then return end
 
+MapVote.Net.receiveWithMiddleware( "MapVote_RequestConfig", function( _, ply )
     net.Start( "MapVote_Config" )
     net.WriteTable( MapVote.GetConfig() )
     net.Send( ply )
-end )
+end, MapVote.Net.requirePermission( MapVote.PermCanConfigure ) )
 
-net.Receive( "MapVote_Config", function( _, ply )
-    if not ply:IsSuperAdmin() then return end
+MapVote.Net.receiveWithMiddleware( "MapVote_Config", function( _, ply )
     local err = MapVote.SetConfig( net.ReadTable() )
     if err ~= nil then
         print( "MapVote: Config is invalid: " .. err )
         -- TODO return this to client
     end
 
     MapVote.SaveConfigToFile( MapVote.defaultConfigFilename )
+end, MapVote.Net.requirePermission( MapVote.PermCanConfigure ) )
+
+net.Receive( "MapVote_ChangeVote", function( _, ply )
```

Comment:
Because this gets rebroadcast to everyone, we should probably have either rate limiting or debounce logic around this receiver.

Reply:
I agree, the original map vote does not have rate limiting, I plan on adding that in a future PR.